pkg/controller/certificate-shim/sync.go

```diff
@@ -140,35 +159,105 @@ func validateIngressTLSBlock(tlsBlock networkingv1beta1.IngressTLS) []error {
 	return errs
 }
 
-func (c *controller) buildCertificates(ctx context.Context, ing *networkingv1beta1.Ingress,
+func validateGatewayListeners(listeners []gatewayapi.Listener) []error {
+	var errs []error
+
+	for _, l := range listeners {
+		if l.TLS == nil {
+			return []error{fmt.Errorf("listener for host %s does not specify a TLS block", *l.Hostname)}
+		}
+		if l.TLS.CertificateRef.Group != "core" {
+			errs = append(
+				errs,
+				fmt.Errorf(
+					"unsupported certificateRef.group, want %s got %s",
+					"core",
+					l.TLS.CertificateRef.Group,
+				),
+			)
+		}
```

Comment:
Should we also look at the TLS block on the `HTTPRoute`? From [this thread](https://github.com/kubernetes-sigs/gateway-api/issues/577#issuecomment-809313624):

![](https://user-images.githubusercontent.com/17914385/112831942-3ffe0300-905a-11eb-9698-d67fc82f0afe.png)

Reply:
👍  I will have to comb over the semantics of gateways and routes and then write a bunch of tests